bin/common.ml

```diff
@@ -1136,6 +1148,17 @@ let init ?action_runner ?log_file c =
         in
         let event = Event.instant ~args common in
         Dune_stats.emit stats event);
+  (* Setup engine hook for printing GC stats to a file *)
+  let () =
+    Dune_engine.Hooks.End_of_build.always @@ fun () ->
```

Comment:
This will run on every iteration of the watch mode. I doubt this is what you intended. Use `at_exit`.

Reply:
Actually I did this intentionally, but now I am not so sure and will change to `at_exit`.